wordpress-mobile/WordPress-iOS
# Recommend App: Implement share app button in About screen

Refs #16995
Closes #16997 
Based on #17006

## Summary
This change adds the share app button to the About screen when the feature flag is enabled. While I did some clean up / tidying for the constants in `AboutViewController`, I tried to keep the implementation/flow as close as possible to the original since this is still feature flagged; once the flag is removed I can take a look at refactoring some parts of the class.

Here's a short video for the implementation:

https://user-images.githubusercontent.com/1299411/129255239-7559120f-9458-4624-b49d-83d13caa8ca6.mov

Notice that the after tapping the button for the first time, the button shows a loading indicator. However, in the later part of the video, the loading indicator no longer shown and the share sheet is displayed instantly. This is intentional, because the presenter is using a cached content and no longer needs to make another roundtrip.

## To Test

- Go to App Settings, and turn on the `Recommend App to Others` feature flag.
- Go to About screen.
- Verify that the share button is shown.
- Tap the share button, and verify that a loading indicator is shown before the share sheet is presented.
- Dismiss the share sheet.
- Tap the share button again, and verify that the loading indicator is no longer shown; the share sheet should be displayed immediately.

## Regression Notes
1. Potential unintended areas of impact
Some parts of the About screen is not displayed correctly. May impact self-hosted / non-WP accounts.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manually tested the screen, with and without the feature flag. Also tested with self-hosted accounts.

3. What automated tests I added (or what prevented me from doing so)
n/a

PR submission checklist:
